plugin/src/main/java/io/fabric8/maven/plugin/mojo/build/ResourceMojo.java

```diff
@@ -270,6 +279,99 @@
 
     private OpenShiftDependencyResources openshiftDependencyResources;
     private OpenShiftOverrideResources openShiftOverrideResources;
+    /**
+     * The artifact type for attaching the generated resource file to the project.
+     * Can be either 'json' or 'yaml'
+     */
+    @Parameter(property = "fabric8.resourceType")
+    private ResourceFileType resourceFileType = yaml;
+    @Component
+    private MavenProjectHelper projectHelper;
+
+    /**
+     * Returns the Template if the list contains a single Template only otherwise returns null
+     */
+    protected static Template getSingletonTemplate(KubernetesList resources) {
+        // if the list contains a single Template lets unwrap it
+        if (resources != null) {
+            List<HasMetadata> items = resources.getItems();
+            if (items != null && items.size() == 1) {
+                HasMetadata singleEntity = items.get(0);
+                if (singleEntity instanceof Template) {
+                    return (Template) singleEntity;
+                }
+            }
+        }
+        return null;
+    }
+
+    public static File writeResourcesIndividualAndComposite(KubernetesList resources, File resourceFileBase, ResourceFileType resourceFileType, Logger log, Boolean generateRoute) throws MojoExecutionException {
+
+        //Creating a new items list. This will be used to generate openshift.yml
+        List<HasMetadata> newItemList = new ArrayList<>();
+
+        if(!generateRoute) {
+
+            //if flag is set false, this will remove the Route resource from resources list
+            for (HasMetadata item : resources.getItems()) {
+                if (item.getKind().equalsIgnoreCase("Route")) {
```

Comment:
Nitpick, I know this is moved from AbstractResourceMojo, but can we replace this with `item instanceof Route`?

Reply:
What is the benefit of using Java types over the Kubernetes/OpenShift type (which is the kind) ?

Reply:
Don't think whether we have any performance benefits. But maybe It looks a bit cleaner?